scripts/operations_tests/operations_tests.py

```diff
@@ -172,6 +177,22 @@ def add_nan(self, image_stack, fraction=0.001):
         data.ravel()[nan_indices] = np.nan
         return image_stack
 
+    def add_flats_and_darks(self, params):
+        for frame_type in ['flat_before', 'flat_after', 'dark_before', 'dark_after']:
+            if frame_type in params:
+                filename_group = FilenameGroup.from_file(config_manager.load_sample)
+                related_filename_group = filename_group.find_related(getattr(FILE_TYPES, params[frame_type]))
+                if related_filename_group:
+                    related_filename_group.find_all_files()
+                    image_stack = loader.load(filename_group=related_filename_group)
+                    if image_stack is not None and hasattr(image_stack, 'data'):
+                        params[frame_type] = image_stack
+                        print(f"Loaded image stack for {frame_type}")
+                    else:
+                        print(f"Failed to load image stack for {frame_type} or image_stack has no 'data' attribute")
+                else:
+                    print(f"Could not find related files for {frame_type}")
```

Comment:
These should raise a `ValueError` as they indicate an issue with the test case or the test data.